src/state/BoardBoss.js

```diff
@@ -0,0 +1,108 @@
+import { ROWS, COLUMNS } from "../constants";
+
+export const SELECTED = "^";
+
+export const EMPTY = "";
+export const PHAROH = "P";
+export const DJED = "D";
+export const OBELISK = "O";
+export const PYRAMID = "P";
+
+export class InvalidSelectionError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = "InvalidSelectionError";
+  }
+}
+
+export class InvalidMoveError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = "InvalidMoveError";
+  }
+}
+
+export default class BoardBoss {
+  constructor() {
+    const buildEmptyBoard = () => {
+      let emptyBoard = [];
+
+      for (let r = 0; r < ROWS; r++) {
+        let newRow = [];
+
+        for (let c = 0; c < COLUMNS; c++) {
+          newRow.push(EMPTY);
+        }
+
+        emptyBoard.push(newRow);
+      }
+
+      return emptyBoard;
+    };
+
+    this.board = buildEmptyBoard();
+  }
+
+  writeSpace(row, column, value) {
+    let newBoard = this.board;
+    newBoard[row][column] = value;
+    this.board = newBoard;
```

Comment:
The final line here is a noop.
```js
this.board = newBoard;
```

The whole thing could be a one-liner. Not sure if you were going for something specific here though.
```ts
this.board[row][column] = value;
```

Reply:
Oh... interesting. I thought I was failing a spec earlier on and though that maybe you couldn't update arrays in place like that in JS (e.g. pass by value vs by reference). Seems like I just had some other issue.